plugin/pkg/admission/gc/gc_admission_test.go

```diff
@@ -191,11 +209,248 @@ func TestGCAdmission(t *testing.T) {
+		Kind:               "ReplicationController",
+		Name:               "rc2",
+		BlockOwnerDeletion: getTrueVar(),
+	}
+	notBlockRC1 := metav1.OwnerReference{
+		APIVersion:         "v1",
+		Kind:               "ReplicationController",
+		Name:               "rc1",
+		BlockOwnerDeletion: getFalseVar(),
+	}
+	notBlockRC2 := metav1.OwnerReference{
+		APIVersion:         "v1",
+		Kind:               "ReplicationController",
+		Name:               "rc2",
+		BlockOwnerDeletion: getFalseVar(),
+	}
+	nilBlockRC1 := metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "ReplicationController",
+		Name:       "rc1",
+	}
+	nilBlockRC2 := metav1.OwnerReference{
+		APIVersion: "v1",
+		Kind:       "ReplicationController",
+		Name:       "rc2",
+	}
+	blockDS1 := metav1.OwnerReference{
+		APIVersion:         "v1",
+		Kind:               "DaemonSet",
+		Name:               "ds1",
+		BlockOwnerDeletion: getTrueVar(),
+	}
+	notBlockDS1 := metav1.OwnerReference{
+		APIVersion:         "v1",
+		Kind:               "DaemonSet",
+		Name:               "ds1",
+		BlockOwnerDeletion: getFalseVar(),
+	}
+	tests := []struct {
+		name     string
+		username string
+		resource schema.GroupVersionResource
+		oldObj   runtime.Object
+		newObj   runtime.Object
+
+		expectedAllowed bool
+	}{
+		// cases for create
+		{
+			name:            "super-user, create, no ownerReferences",
+			username:        "super",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(),
+			expectedAllowed: true,
+		},
+		{
+			name:            "super-user, create, all ownerReferences have blockOwnerDeletion=false",
+			username:        "super",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(notBlockRC1, notBlockRC2),
+			expectedAllowed: true,
+		},
+		{
+			name:            "super-user, create, some ownerReferences have blockOwnerDeletion=true",
+			username:        "super",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(blockRC1, blockRC2),
+			expectedAllowed: true,
+		},
+		{
+			name:            "non-rc-deleter, create, no ownerReferences",
+			username:        "non-rc-deleter",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(),
+			expectedAllowed: true,
+		},
+		{
+			name:            "non-rc-deleter, create, all ownerReferences have blockOwnerDeletion=false or nil",
+			username:        "non-rc-deleter",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(notBlockRC1, nilBlockRC2),
+			expectedAllowed: true,
+		},
+		{
+			name:            "non-rc-deleter, create, some ownerReferences have blockOwnerDeletion=true",
+			username:        "non-rc-deleter",
+			resource:        api.SchemeGroupVersion.WithResource("pods"),
+			newObj:          podWithOwnerRefs(blockRC1, notBlockRC2),
+			expectedAllowed: false,
```

Comment:
We have multiple deny branches now.  Please check the error message to make sure it's the failure you're expecting.